### ISSUE TITLE
cpplint flags cstdio to be missing when stdio.h is included

### ISSUE THREAD

**Issue by pkern** [cpplint/cpplint#302]
In https://github.com/google/glome/actions/runs/12331876182/job/34419000483 cpplint is complaining about cstdio not being included for various C functions that are in fact being included through stdio.h. Is there a way to specify C mode instead of C++ mode for files? I could not find such a function (and it complaining about this is new).

**Comment by hongbo-miao** [comment]
Got same error

```sh
main.c:4:  Add #include <cstdio> for printf  [build/include_what_you_use] [4]
Done processing main.c
```

for a C file (not C++)

main.c
```c
#include <stdio.h>

int main(void) {
  printf("Hello, world!\n");
  return 0;
}
```

**Comment by notjl** [comment]
Got the same error on my nvim setup for c

![Image](https://github.com/user-attachments/assets/af1b3dd4-3fb4-4599-b75e-98b550d989da)
